odoo/fields.py

```diff
@@ -3047,6 +3058,518 @@ def _record_ids_per_res_model(self, records):
+
+            elif property_type == 'many2many' and is_list_of(property_value, int):
+                property_value = env[property_model].browse(property_value).name_get()
+                property_definition[value_key] = property_value or False
+
+        return values_list
+
+    @classmethod
+    def _validate_values(cls, values_list, value_key='value'):
+        """Remove any incoherent properties values.
+
+        :param values_list: List of properties definition with value
+        :param value_key: The dict key which contains the values to validate
+        """
+        for values in values_list:
+            property_value = values.get(value_key)
+            if not property_value:
+                continue
+
+            property_type = values.get('type')
+            if property_type == 'many2one' and not isinstance(property_value, int):
+                values[value_key] = False
+
+            elif property_type == 'many2many':
+                if not is_list_of(property_value, int):
+                    values[value_key] = False
+                elif len(property_value) != len(set(property_value)):
+                    # remove duplicated value and preserve order
+                    values[value_key] = list(dict.fromkeys(property_value))
+
+        return values_list
+
+    @classmethod
+    def _parse_json_types(cls, values_list, env):
+        """Parse the value stored in the JSON.
+
+        Check for records existence, if we removed a selection option,...
+
+        :param values_list: List of properties definition and values
+        :param env: environment
+        """
+        for property_definition in values_list:
+            property_value = property_definition.get('value')
+            property_type = property_definition.get('type')
+            current_model = property_definition.get('model')
+
+            assert property_type in cls.ALLOWED_TYPES, f'Wrong property type {property_type!r}'
+
+            if property_type == 'many2one' and property_value:
+                # [1337, 'res.partner']
+                assert is_list(property_value, [int, str]), f'Wrong many2one value: {property_value!r}.'
+
+                res_id, res_model = property_value
+                if not current_model or res_model != current_model:
+                    # the model on the child does not match the model on the parent
+                    # maybe we have changed the model on the parent, we should reset
+                    # the many2one value
+                    property_value = False
+                else:
+                    # many2one might have been deleted
+                    property_value = env[res_model].browse(res_id).exists().id
+
+            elif property_type == 'many2many' and property_value:
+                # [[1337, 1338, 1339], 'res.partner']
+                assert is_list(property_value, [(list, tuple), str]), f'Wrong many2many value: {property_value!r}.'
```

Comment:
Not sure I see a tuple in your example here above.

EDIT oh wait, it wants a list OR a tuple?